### ISSUE TITLE
How can I match detected objects name and its data

### ISSUE THREAD

**Issue by gurselturkeri** [ultralytics/ultralytics#3864]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hi, I would like to match detected object class name and its coordinates. When YOLO detected more objects I can not match its class and coordinates. How can I match detected objects and its data.

![Screenshot from 2023-07-21 09-58-11](https://github.com/ultralytics/ultralytics/assets/68682737/5d30f635-167f-4042-bcbc-49d62305e9f7)

**Comment by xaristeidou** [comment]
@gurselturkeri What you actually have to do is to match the position of the predicted class number and the position of the xyxy coordinates. You can do it with the following code:

```
from ultralytics import YOLO
import cv2

model = YOLO("yolov8m.pt")

results = model.predict(source = "video.mp4", imgsz = 640, conf = 0.35, show = True, save = False)


for i, obj in enumerate(results[0].boxes.cls):
    object_class_name = results[0].names[results[0].boxes.cls[i].item()]
    x1, y1, x2, y2 = results[0].boxes.xyxy[i]
    x1, y1, x2, y2 = int(x1.item()), int(y1.item()), int(x2.item()), int(y2.item())
    print(object_class_name, x1, y1, x2, y2)
```

Let's see what actually happens in the code step by step:

Here we use a video sample "video.mp4" in which we predict. 

`results[0]` returns the predictions of the first frame of the video.

`results[0].boxes.cls` return the class number for each prediction in a torch tensor. The result look like this:
`tensor([2., 2., 2., 2., 2., 2., 2., 2., 2., 2., 2., 2., 2., 2., 2., 2., 2., 2., 7., 2., 7., 2., 2., 2., 2., 2., 2., 2., 2., 2., 2., 2.], device='cuda:0')`

`results[0].boxes.xyxy` return the x1,y1,x2,y2 coordinates for each of the detected objects in a torch tensor. You can choose any position to retrieve the coordinates for a specific object. e.g. `results[0].boxes.xyxy[0]` returns the coordinates for the first predicted object. `tensor([776.2866, 499.9119, 905.7246, 633.5973], device='cuda:0')`

So what is actually happening here is the each position of each tensor corresponds to the coordinates of the object to the xyxy tensor.

`results[0].names` contains in a dictionary the number of each class and its' corresponding name, so we use this to extract the name of the class using the predicted number of class.
For example `results[0].names[2.0]` return `"car"`

To sum up we then run a for loop with enumerate object in order to get the x,y coordinates for each detected object and print the name of the object and its' corresponding coordinates.

If you want to predict on a video sample then to iterate in each frame of the video you have to use one more `for` loop:
`for frame in results:
    for i, obj in enumerate(frame.boxes.cls):
        object_class_name = results[0].names[results[0].boxes.cls[i].item()]
        x1, y1, x2, y2 = results[0].boxes.xyxy[i]
        x1, y1, x2, y2 = int(x1.item()), int(y1.item()), int(x2.item()), int(y2.item())
        print(object_class_name, x1, y1, x2, y2)`


Otherwise if you want to predict on a streaming loop you use results[0] because for each frame of the webcam there is only one single frame to predict.

**Comment by glenn-jocher** [comment]
@gurselturkeri in YOLOv8, when an object is detected, the detection result includes class labels and bounding box coordinates. The class labels and bounding box coordinates are maintained in the same order - meaning the first class label corresponds to the first set of bounding box coordinates, the second class label corresponds to the second set of bounding box coordinates, and so on. 

This means that if you iterate over these two lists simultaneously - you would be able to match the detected object's class name with its corresponding bounding box coordinates.

For example, if the detected class list is ['person', 'car', 'dog'] and the bounding box list is [(x1, y1, x2, y2), (x1, y1, x2, y2), (x1, y1, x2, y2)], then 'person' corresponds to the first set of coordinates, 'car' corresponds to the second set of coordinates, and 'dog' corresponds to the third set of coordinates.

To specifically connect the exact class name with its corresponding bounding box coordinates, simply use the index while iterating through the list of detections (both the class list and bounding box list). 

This ensures that for every detected object, you're correctly assigning the matching class name and bounding box coordinates.
